llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -2162,19 +2162,20 @@ AArch64TTIImpl::getRegisterBitWidth(TargetTransformInfo::RegisterKind K) const {
   case TargetTransformInfo::RGK_Scalar:
     return TypeSize::getFixed(64);
   case TargetTransformInfo::RGK_FixedWidthVector:
-    if (!ST->isNeonAvailable() && !EnableFixedwidthAutovecInStreamingMode)
-      return TypeSize::getFixed(0);
-
-    if (ST->hasSVE())
+    if (ST->useSVEForFixedLengthVectors() &&
+        (ST->isSVEAvailable() || EnableFixedwidthAutovecInStreamingMode))
       return TypeSize::getFixed(
           std::max(ST->getMinSVEVectorSizeInBits(), 128u));
```

Comment:
I'm not sure this is all that safe, although appreciate the StreamingSVE side of the functionality is only for testing.  I say this because the functionality is driven by `vscale_range` which is itself driven by `-msve-vector-bits`, both of whom knew nothing about SME at the point of implementation.

How do you feel about extending this PR to update clang so that `-msve-vector-bits` does not affect the `vscale_range` attribute of streaming (perhaps even streaming compatible) functions? That way streaming functions will be capped at the safe value of 128-bit.

Reply:
Alternatively you could implement `-mssve-vector-bits` to allow both quantities to be set independently and then only allow it to be use for streaming compatible functions when the both are specified and equal.  Perhaps I've talked myself into this needing a separate PR :)

Reply:
Yes we likely need to think about how the feature should work in the face of SME or at least add a clarification about potential undefined behaviour when the SVE and SSVE vector lengths do not match.